### PR TITLE
statistics: add default value of CMSketch for `Analyze` (#19455)

### LINKED ISSUE #19343: statistic: the default value of CMSketch is not initialized correctly when using `Analyze`
## Development Task
How to reproduce:
```
create table t (a int, key(a));
insert into t values (0);
insert into t values (0);
...
insert into t values (0); // insert 2048 rows

insert into t values (1);
insert into t values (2);
insert into t values (3);

analyze table t with 0 topn;
explain select * from t where a = 1;
explain select * from t where a = 2;
explain select * from t where a = 3;
```

The estimated rows for `a = 1/2/3` is zero, which is not accurate.

This is a bad case for the heuristic rule of removing noise when estimating point get:
```
		noise := (c.count - uint64(c.table[i][j])) / (uint64(c.width) - 1)
		if uint64(c.table[i][j]) < noise {
			vals[i] = 0
		} else {
			vals[i] = c.table[i][j] - uint32(noise)
		}
```

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #19455 to release-4.0

---

### What problem does this PR solve?

Issue Number: close #19343 

Problem Summary:
Add default value of CMSketch for `Analyze`

### What is changed and how it works?
We use `count / NDV` in CMSketch as default value. This means count and NDV are not include topN.

Proposal: [xxx](url) 

What's Changed:
- Use `count / NDV` as default value.
- In `queryHashValue`, if `res` is 0 before the noise is eliminated, the default value is not used.

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- 
- Add default value of CMSketch for Analyze

### CODE REVIEW

**Review by zz-jason — COMMENTED** [review]
LGTM

File: executor/analyze_test.go [hunk 1]
```diff
@@ -592,3 +592,82 @@ func (s *testSuite1) TestHashInTopN(c *C) {
...
+		"test t3  b 0 0 1 1 1 1",
+		"test t3  b 0 1 2 1 2 2",
+		"test t3  b 0 2 3 1 3 3",
+		"test t3  c 0 0 1 1 1 1",
+		"test t3  c 0 1 2 1 2 2",
+		"test t3  c 0 2 3 1 3 3",
+		"test t3  c 1 0 1 1 1 1",
+		"test t3  c 1 1 2 1 2 2",
+		"test t3  c 1 2 3 1 3 3"))
+}
```
Comment (qw4990) [inline executor/analyze_test.go:641]: Should we remove this test since it doesn't belong to your PR https://github.com/pingcap/tidb/pull/19455? @rebelice

Comment (rebelice) [reply executor/analyze_test.go:641]: Yes, I'll handle it.

**Review by qw4990 — APPROVED** [review]
LGTM

**Pull request merged**